be/src/storage/rowset/segment_v2/bitshuffle_page.h

```diff
@@ -356,6 +370,51 @@ class BitShufflePageDecoder final : public PageDecoder {
         return Status::OK();
     }
 
+    Status save_in_page_cache(PageCacheOptions* opts) override {
+        DCHECK(_parsed);
+        if (!opts->save_in_page_cache || _data_from_cache) {
+            return Status::OK();
+        }
+
+        auto cache = StoragePageCache::instance();
+        PageCacheHandle cache_handle;
+        StoragePageCache::CacheKey cache_key(opts->rblock->path(), opts->page_pointer.offset);
+        Slice page_slice = _options.page_handle->data();
+
+        uint32_t footer_size = decode_fixed32_le((uint8_t*)page_slice.data + page_slice.size - 4);
+        size_t data_size = _num_element_after_padding * _size_of_element;
+        size_t header_size = page_slice.size - 4 - footer_size - opts->nullmap_size -
+                             (_compressed_size - BITSHUFFLE_PAGE_HEADER_SIZE);
+
+        std::unique_ptr<char[]> decompressed_page(
+                new char[page_slice.size + data_size - (_compressed_size - BITSHUFFLE_PAGE_HEADER_SIZE)]);
+        memcpy(decompressed_page.get(), page_slice.data, header_size);
+        Slice compressed_body(page_slice.data + header_size, _compressed_size - BITSHUFFLE_PAGE_HEADER_SIZE);
+        Slice decompressed_body(&(decompressed_page.get()[header_size]), data_size);
+        int64_t bytes = bitshuffle::decompress_lz4(compressed_body.data, decompressed_body.data,
+                                                   _num_element_after_padding, _size_of_element, 0);
+        if (bytes != compressed_body.size) {
+            return Status::Corruption(
+                    strings::Substitute("decompress failed: expected number of bytes consumed=&0 vs real consumed=$1",
+                                        compressed_body.size, bytes));
+        }
+        memcpy(decompressed_body.data + decompressed_body.size,
+               page_slice.data + header_size + (_compressed_size - BITSHUFFLE_PAGE_HEADER_SIZE),
+               opts->nullmap_size + footer_size + 4);
+
+        Slice cache_slice =
+                Slice(decompressed_page.get(), header_size + data_size + opts->nullmap_size + footer_size + 4);
+        cache->insert(cache_key, cache_slice, &cache_handle, opts->kept_in_memory);
```

Comment:
invalid usage of PageCacheHandle

Reply:
> 

fix in https://github.com/StarRocks/starrocks/pull/1597/commits/476bbd7e7d18e096b091abc0c6f4129cb5f94a68, PTAL